codec.go

```diff
@@ -175,7 +191,11 @@ func (u *UUID) decodeURN(t []byte) error {
 	urnUUIDPrefix := t[:9]
 
 	if !bytes.Equal(urnUUIDPrefix, urnPrefix) {
-		return fmt.Errorf("uuid: incorrect UUID format in string %q", t)
+		// Copying t will prevent the Go compiler from allocating t on
+		// the heap if t would otherwise be on the stack.
+		tc := make([]byte, len(t))
```

Comment:
I believe this should be handled by compiler optimizations, rather than code changes (especially if the code is identical in all cases).